lib/improver/tests/ensemble_calibration/ensemble_calibration/test_EstimateCoefficientsForEnsembleCalibration.py

```diff
@@ -154,24 +165,68 @@ def test_statsmodels_realizations(self, warning_list=None):
         distribution = "gaussian"
         desired_units = "degreesC"
         predictor_of_mean_flag = "realizations"
-        no_of_realizations = 3
-        estimate_coefficients_from_linear_model_flag = True
 
         if not statsmodels_found:
-            plugin = Plugin(distribution, desired_units,
-                            predictor_of_mean_flag=predictor_of_mean_flag)
+            Plugin(distribution, desired_units,
+                   predictor_of_mean_flag=predictor_of_mean_flag)
             warning_msg = "The statsmodels can not be imported"
             self.assertTrue(any(item.category == ImportWarning
                                 for item in warning_list))
             self.assertTrue(any(warning_msg in str(item)
                                 for item in warning_list))
 
 
+class Test__repr__(IrisTest):
+
+    """Test the __repr__ method."""
+
+    @ManageWarnings(
+        ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
+    def setUp(self):
+        """Set up values for tests."""
+        self.distribution = "gaussian"
+        self.current_cycle = "20171110T0000Z"
+        self.minimiser = repr(ContinuousRankedProbabilityScoreMinimisers())
+        self.coeff_names = ["gamma", "delta", "alpha", "beta"]
+
+    @ManageWarnings(
+        ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
+    def test_basic(self):
+        """Test without specifying keyword arguments"""
+        result = str(Plugin(self.distribution, self.current_cycle))
+        msg = ("<EstimateCoefficientsForEnsembleCalibration: "
+               "distribution: gaussian; "
+               "current_cycle: 20171110T0000Z; "
+               "desired_units: None; "
+               "predictor_of_mean_flag: mean; "
+               "minimiser: <class 'improver.ensemble_calibration."
+               "ensemble_calibration."
+               "ContinuousRankedProbabilityScoreMinimisers'>; "
+               "coeff_names: ['gamma', 'delta', 'alpha', 'beta']>")
+        self.assertEqual(result, msg)
+
+    @ManageWarnings(
+        ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
+    def test_with_kwargs(self):
+        """Test when keyword arguments are specified."""
+        result = str(Plugin(
+            self.distribution, self.current_cycle,
+            desired_units="Kelvin", predictor_of_mean_flag="realizations"))
+        msg = ("<EstimateCoefficientsForEnsembleCalibration: "
+               "distribution: gaussian; "
+               "current_cycle: 20171110T0000Z; "
+               "desired_units: Kelvin; "
+               "predictor_of_mean_flag: realizations; "
+               "minimiser: <class 'improver.ensemble_calibration."
+               "ensemble_calibration."
+               "ContinuousRankedProbabilityScoreMinimisers'>; "
+               "coeff_names: ['gamma', 'delta', 'alpha', 'beta']>")
+        self.assertEqual(result, msg)
+
+
 class Test_create_coefficients_cube(IrisTest):
```

Comment:
So we need a test to check what happens if historic forecasts forecast period coord does not all have the same value? Or is this covered elsewhere?

Reply:
I've added a unit test to cover this.